fix_diff.py

```
@@ -0,0 +1,44 @@
+            if m:
+                public_methods.add(m.group(1))
+                
+    # Now patch curr
+    out = []
+    for line in curr:
+        if line.startswith("    ") and not line.startswith("    public ") and not line.startswith("    //") and not line.startswith("    /*") and not line.startswith("    *"):
+            m = re.search(r'^    (?:[\w<>,\[\] ]+)\s+(\w+)\s*\(', line)
+            if m:
+                method_name = m.group(1)
+                if method_name in public_methods:
+                    line = line.replace("    ", "    public ", 1)
```

Ruff (0.15.12)

[warning] `for` loop variable `line` overwritten by assignment target

(PLW2901)